presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -533,4 +565,37 @@ public Boolean process(Node node, @Nullable Void context)
             return super.process(node, context);
         }
     }
+
+    /**
+     * Verifies that no query node output columns are referenced from ORDER BY expressions.
+     * Only FROM columns are allowed within ORDER BY expressions.
+     */
+    private class VerifyOrderByAggregationVisitor
+            extends DefaultTraversalVisitor<Void, Void>
+    {
+        @Override
+        protected Void visitIdentifier(Identifier node, Void context)
+        {
+            verifyField(node, QualifiedName.of(node.getName()));
+            return null;
+        }
+
+        @Override
+        protected Void visitDereferenceExpression(DereferenceExpression node, Void context)
+        {
+            if (analysis.getColumnReferences().contains(node)) {
+                verifyField(node, DereferenceExpression.getQualifiedName(node));
+            }
+
+            return null;
+        }
+
+        private void verifyField(Expression node, QualifiedName qualifiedName)
+        {
+            ResolvedField resolvedField = orderByScope.get().resolveField(node, qualifiedName);
```

Comment:
applied comment (`return null`), but still you have to add tests in: https://github.com/prestodb/presto/issues/7030